examples/capable/src/main.rs

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+// Copyright (c) 2021 BMC Software, Inc.
+// Author Devasia Thomas <https://www.linkedin.com/in/devasiathomas/>
+//
+// Based on capable(8) by Brendan Gregg
+use core::time::Duration;
+use std::str::FromStr;
+
+use anyhow::{bail, Result};
+use chrono::Local;
+use libbpf_rs::PerfBufferBuilder;
+use phf::phf_map;
+use plain::Plain;
+use structopt::StructOpt;
+
+#[path = "bpf/.output/capable.skel.rs"]
+mod capable;
+
+use capable::capable_rodata_types::Unique;
+use capable::*;
+
+static CAPS: phf::Map<i32, &'static str> = phf_map! {
+    0i32 => "CAP_CHOWN",
+    1i32 => "CAP_DAC_OVERRIDE",
+    2i32 => "CAP_DAC_READ_SEARCH",
+    3i32 => "CAP_FOWNER",
+    4i32 => "CAP_FSETID",
+    5i32 => "CAP_KILL",
+    6i32 => "CAP_SETGID",
+    7i32 => "CAP_SETUID",
+    8i32 => "CAP_SETPCAP",
+    9i32 => "CAP_LINUX_IMMUTABLE",
+    10i32 => "CAP_NET_BIND_SERVICE",
+    11i32 => "CAP_NET_BROADCAST",
+    12i32 => "CAP_NET_ADMIN",
+    13i32 => "CAP_NET_RAW",
+    14i32 => "CAP_IPC_LOCK",
+    15i32 => "CAP_IPC_OWNER",
+    16i32 => "CAP_SYS_MODULE",
+    17i32 => "CAP_SYS_RAWIO",
+    18i32 => "CAP_SYS_CHROOT",
+    19i32 => "CAP_SYS_PTRACE",
+    20i32 => "CAP_SYS_PACCT",
+    21i32 => "CAP_SYS_ADMIN",
+    22i32 => "CAP_SYS_BOOT",
+    23i32 => "CAP_SYS_NICE",
+    24i32 => "CAP_SYS_RESOURCE",
+    25i32 => "CAP_SYS_TIME",
+    26i32 => "CAP_SYS_TTY_CONFIG",
+    27i32 => "CAP_MKNOD",
+    28i32 => "CAP_LEASE",
+    29i32 => "CAP_AUDIT_WRITE",
+    30i32 => "CAP_AUDIT_CONTROL",
+    31i32 => "CAP_SETFCAP",
+    32i32 => "CAP_MAC_OVERRIDE",
+    33i32 => "CAP_MAC_ADMIN",
+    34i32 => "CAP_SYSLOG",
+    35i32 => "CAP_WAKE_ALARM",
+    36i32 => "CAP_BLOCK_SUSPEND",
+    37i32 => "CAP_AUDIT_READ",
+    38i32 => "CAP_PERFMON",
+    39i32 => "CAP_BPF",
+    40i32 => "CAP_CHECKPOINT_RESTORE",
+};
+
+impl FromStr for Unique {
+    type Err = &'static str;
+    fn from_str(unq_type: &str) -> Result<Self, Self::Err> {
+        match unq_type {
+            "0" => Ok(Unique::UnqOff),
+            "1" => Ok(Unique::UnqPid),
+            "2" => Ok(Unique::UnqCgroup),
+            _ => Err("Use 1 for pid (default), 2 for cgroups"),
+        }
+    }
+}
+
+/// Trace capabilities
+#[derive(Debug, Copy, Clone, StructOpt)]
+#[structopt(name = "examples", about = "Usage instructions")]
+struct Command {
+    /// verbose: include non-audit checks
+    #[structopt(short, long)]
+    verbose: bool,
+    /// only trace <pid>
+    #[structopt(short, long, default_value = "0")]
+    pid: u32,
+    /// extra fields: Show TID and INSETID columns
+    #[structopt(short = "x", long = "extra")]
+    extra_fields: bool,
+    /// don't repeat same info for the same pid<1> or cgroup<2>
+    #[structopt(long = "unique", default_value = "0")]
+    unique_type: Unique,
```

Comment:
Is the bcc tool using integers as part of cmd line options? If not, would be better to do something more easily human readable

Reply:
I don't think the --unique worked as advertised in the bcc tool. Thats why I did this. I can definitely change it to parse strings. no probs.